hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/shell/TestBulkDeleteCommand.java

```diff
@@ -64,4 +78,52 @@ public void testArguments() throws IOException, URISyntaxException {
         Assertions.assertThat(bulkDeleteCommand.childArgs.get(0)).
                 describedAs("Children arguments must match").isEqualTo(arg2);
     }
+
+    @Test
+    public void testLocalFileDeletion() throws IOException {
+        String deletionDir = "toDelete";
+        String baseFileName = "file_";
+        Path baseDir = new Path(testRootDir, deletionDir);
+        List<String> listOfPaths = new ArrayList<>();
+        for(int i = 0; i < 100; i++) {
+            Path p = new Path(baseDir, baseFileName + i);
+            lfs.create(p);
+            listOfPaths.add(p.toUri().toString());
+        }
+        List<String> finalCommandList = new ArrayList<>();
+        finalCommandList.add("-bulkDelete");
+        finalCommandList.add(baseDir.toUri().toString());
+        finalCommandList.addAll(listOfPaths);
+        shell.run(finalCommandList.toArray(new String[0]));
+        Assertions.assertThat(lfs.listFiles(baseDir, false).hasNext())
+                .as("All the files should have been deleted").isEqualTo(false);
+
+    }
+
+    @Test
+    public void testLocalFileDeletionWithFileName() throws IOException {
+        String deletionDir = "toDelete";
+        String baseFileName = "file_";
+        Path baseDir = new Path(testRootDir, deletionDir);
+        Path fileWithDeletePaths = new Path(testRootDir, "fileWithDeletePaths");
+        FSDataOutputStream fsDataOutputStream = lfs.create(fileWithDeletePaths, true);
+        BufferedWriter br = new BufferedWriter(new OutputStreamWriter(fsDataOutputStream));
+        for(int i = 0; i < 100; i++) {
+            Path p = new Path(baseDir, baseFileName + i);
+            lfs.create(p);
+            br.write(p.toUri().toString());
+            br.newLine();
+        }
+        br.flush(); // flush the file to write the contents
+        br.close(); // close the writer
+        List<String> finalCommandList = new ArrayList<>();
+        finalCommandList.add("-bulkDelete");
+        finalCommandList.add("-readFromFile");
+        finalCommandList.add(fileWithDeletePaths.toUri().toString());
+        finalCommandList.add(baseDir.toUri().toString());
+        shell.run(finalCommandList.toArray(new String[0]));
+        Assertions.assertThat(lfs.listFiles(baseDir, false).hasNext())
+                .as("All the files should have been deleted").isEqualTo(false);
```

Comment:
add "should have been deleted under the <given path>"

Reply:
Done